src/librustc/middle/check_match.rs

```diff
@@ -864,7 +864,10 @@ pub fn specialize<'a>(cx: &MatchCheckCtxt, r: &[&'a Pat],
                 } else {
                     None
                 },
-                _ => Some(repeat(DUMMY_WILD_PAT).take(arity).collect())
+                _ => match *constructor {
+                    ConstantValue(..) => Some(vec![]),
```

Comment:
so I'm not totally familiar with this code; I'm looking at what you've added here, and I do not immediately see why it suffices to add a special case for `ConstantValue(..)` but not need one for e.g. `ConstantRange` .  Can you enlighten me?